app/models/mentee_application_state.rb

```diff
@@ -0,0 +1,42 @@
+# == Schema Information
+#
+# Table name: mentee_application_states
+#
+#  id                         :bigint           not null, primary key
+#  note                       :text
+#  status                     :integer
+#  created_at                 :datetime         not null
+#  updated_at                 :datetime         not null
+#  status_changed_by_id       :integer
+#  user_mentee_application_id :bigint           not null
+#
+# Indexes
+#
+#  index_mentee_application_states_on_user_mentee_application_id  (user_mentee_application_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (user_mentee_application_id => user_mentee_applications.id)
+#
+class MenteeApplicationState < ApplicationRecord
+  belongs_to :user_mentee_application
+
+  STATUSES = {
+    pending: 0,
+    stage_one: 1,
+    stage_two: 2,
+    stage_three: 3,
+    stage_four: 4,
+    accepted: 5,
+    rejected: 6
+  }.freeze
+
+  enum status: STATUSES
```

Comment:
You may already know this, but you don't have to write it this way to be able to refer to the enum hash
```rb
enum status: {
  pending: 0,
  stage_one: 1,
  # etc.
}
  
MenteeApplication.statuses
#=> { "pending" => 0, "stage_one" => 1 } # etc.

# the returned hash also has 'indifferent access', which means you can key in with strings or symbols
MenteeApplication.statuses["pending"] #=> 0
MenteeApplication.statuses[:pending] #=> 0
```
Just some information if you didn't know.

Reply:
hm, I figured it would be good to have an independent constant with different statuses, but now I'm wondering if it's just repeating code?

Reply:
what Josh said is correct and that is the general approach.